### PR TITLE
Linting and Formatting for the backend

### PR DESCRIPTION
**Pull request by ggggg**
# Fixes #
Issue #62 

## Proposed Changes

  - Adds linting to the backend
  - Adds linting CI using github actions

## Tests

Please describe the tests that you ran to verify your changes.

- [x] Compiled typescript via `yarn run build` to make sure no type errors are caused
- [x] Ran eslint style checks

### CODE REVIEW

**Comment by ggggg** [comment]
Note: This PR is missing husky integration

**Comment by hiimchrislim** [comment]
If possible, could you also add in the Husky integration with this? Thanks!

**Comment by ggggg** [comment]
> If possible, could you also add in the Husky integration with this? Thanks!

No because I cant get it to work on windows, this could be a separate issue.

**Comment by shubhbapna** [comment]
Husky with github desktop seems to be a known issue with some working hacks

Try:
https://github.com/typicode/husky/issues/1163
https://stackoverflow.com/questions/66520423/github-desktop-husky-pre-commit-hook-does-not

File: server/package.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
     "prestart": "tsc",
     "start": "node .",
     "dev": "nodemon --watch src -e ts,ejs --exec yarn run start",
-    "build": "tsc --project ./"
+    "build": "tsc --project ./",
+    "lint": "eslint --ext .ts --fix ."
```
Comment (shubhbapna) [inline server/package.json]: Add 2 scripts:
- `lint`: no `fix` option
- `lint:fix`: with `fix` passed as option

**Review by shubhbapna — CHANGES_REQUESTED** [review]

File: server/package.json (hunk 1 above)
Comment (ggggg) [reply server/package.json]: Done

**Comment by hiimchrislim** [comment]
> No because I cant get it to work on windows, this could be a separate issue.


What errors are you encountering with Husky on Windows?

**Comment by ggggg** [comment]
Have a look at https://discord.com/channels/732778344466087967/1025580306926948494/1029134370671427584

**Review by shubhbapna — APPROVED** [review]
LGTM

**Comment by shubhbapna** [comment]
For next time please try to resolve such issues

You will definitely come across projects where you might not be able to setup your dev environment as smoothly. But it is definitely worth setting up your dev env correctly otherwise your PRs might be blocked until their standards are met.

But congrats on your first PR!

**Pull request merged**